design-proposals/dynamic-primary-pod-nic-name.md

```diff
@@ -527,3 +531,34 @@ The primary pod interface name is set by the container runtime (for example cri-
 
 - There is no K8s API to name the pod's primary NIC
 - There is no Multus API or config to name the pod's primary NIC
+
+# Appendix B - Extending support for iface change
+
+An advanced scenario in which during a migration, the pod interface name
+changes between the source and target, is an edge case which we have intentionally
+not treated (mentioned as non-goal).
+
+However, concerns have been raised about a future change that might require
+a re-evaluation of the limitation, allowing such name changes.
+
+Therefore, a possible solution is outlined here, showing that it is possible
+to extend this proposal and add such support.
+
+## Solution
+
+The suggested solution is based on the VMI [migration status](https://kubevirt.io/user-guide/compute/live_migration/#migration-status).
+
+A new field may be added to the migration status, indicating the target pod interface name.
+The Migration controller can update the field based on the target pod network-status annotation.
+
+On the source, the virt-handler and virt-launcher components will continue
```

Comment:
The source virt-launcher will probably need to edit the domain xml prior to sending it to the target, as is done in other similar features.

Reply:
The iface name is not specified on the domain xml, except for rare cases. And that problem is not related to this specific feature.

Reply:
For domain attachments of type `tap` we fill the domain's interface's target.Device field [1] with the name of the tap. We derive the tap's name from the pod interface name [2][3].

Since the target pod could have a pod interface name that is different from the source - the tap name in the target will be different as well.

Other features, such as dedicated CPUs[4] modify the domain in the source, with info collected from the target prior to sending it.

[1] https://libvirt.org/formatdomain.html#generic-ethernet-connection
[2] https://github.com/kubevirt/kubevirt/blob/f95b08230d92ea5ec0603164b943c27ff0f90899/pkg/network/setup/netpod/netpod.go#L384
[3] https://github.com/kubevirt/kubevirt/blob/917ea8e368e6e9816e656be7883abaf3e24ad27e/pkg/network/domainspec/generators.go#L102
[4] https://github.com/kubevirt/kubevirt/blob/22b1872d965c2c39885ddee2542a42ce12b4da49/pkg/virt-launcher/virtwrap/live-migration-source.go#L219
[5]

Reply:
The information itself is still available on the vmi status for the target and source, so I do not see a problem to make it work.

I have intentionally avoided to give all the details as this requires more work with hopefully not a big chance to occur in the future.

I am unsure how opening this up to low level details contributes to getting this in and with it to GA the feature in general.

Reply:
I missed the fact that the interface name is derived from the tap and is set in the domain XML...

Reply:
I think we must find a way to mitigate this issue. Perhaps we need to implement a way to communicate and use the target tapdevice to the source via the vmi.status.migrationState

Reply:
The route that I see right now is to follow the same synchronization mechanism we have in place for re-adjusting the CPU affinity of the domain when migrating to the target.
As in https://github.com/kubevirt/kubevirt/blob/main/pkg/virt-launcher/virtwrap/live-migration-source.go#L148

The idea is, as before, to let the migration controller update the primary target interface (only if it's different from the source) in the `vmi.status.migrationState`
Then adjust the target interface in the migratable domain XML 
https://github.com/kubevirt/kubevirt/blob/main/pkg/virt-launcher/virtwrap/live-migration-source.go#L199

I'm not 100% sure what else is needed to be done in the virt-launcher to fully support the change.
If I'm missing anything in my view, let's have a call to discuss it.

Reply:
I prefer not to have a solution where the domain xml is changed on source.
We will need always to think and implement upfront for the unknown future.
We already avoided using this technique in the past when the pod ifaces changed from ordered to hashed (for the hotplug/unplug support).